src_tests/unittests/osm_configurator/view/states/test_state_manager.py

```diff
@@ -0,0 +1,184 @@
+
+
+def test_change_state_category(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.CATEGORY)
+
+    assert state_manager.get_state().get_state_name() == state_name_enum_i.StateName.CATEGORY
+
+
+def test_change_state_reduction(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.REDUCTION)
+
+    assert state_manager.get_state().get_state_name() == state_name_enum_i.StateName.REDUCTION
+
+
+def test_change_state_attractivity_edit(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.ATTRACTIVITY_EDIT)
+
+    assert state_manager.get_state().get_state_name() == state_name_enum_i.StateName.ATTRACTIVITY_EDIT
+
+
+def test_change_state_attractivity_view(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.ATTRACTIVITY_VIEW)
+
+    assert state_manager.get_state().get_state_name() == state_name_enum_i.StateName.ATTRACTIVITY_VIEW
+
+
+def test_change_state_aggregation(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.AGGREGATION)
+
+    assert state_manager.get_state().get_state_name() == state_name_enum_i.StateName.AGGREGATION
+
+
+def test_change_state_calculation(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.CALCULATION)
+
+    assert state_manager.get_state().get_state_name() == state_name_enum_i.StateName.CALCULATION
+
+
+def test_change_state_settings_project(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.SETTINGS_PROJECT)
+
+    assert state_manager.get_state().get_state_name() == state_name_enum_i.StateName.SETTINGS_PROJECT
+
+
+def test_change_state_settings_no_project(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.SETTINGS_NO_PROJECT)
+
+    assert state_manager.get_state().get_state_name() == state_name_enum_i.StateName.SETTINGS_NO_PROJECT
+
+def test_default_go_left_and_right_none(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
+
+    state_manager.change_state(state_name_enum_i.StateName.MAIN_MENU)
+
+    # The MainMenu hast no default left or right
+
+    assert not state_manager.default_go_left()
+
+    assert not state_manager.default_go_right()
+
+def test_default_go_left_and_rigth(main_window):
+    state_manager: state_manager_i.StateManager = main_window._state_manager
```

Comment:
Wouldn't it be better to declare the state manager once and then reuse it everytime?

Reply:
I think it's good, that the tests are independent af each other

Reply:
what karl said